src/base/net/reverseproxy.h

```diff
@@ -0,0 +1,50 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2018 Eric Prévoteau <eric.prevoteau@ac-strasbourg.fr>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#ifndef NET_REVERSEPROXY
+#define NET_REVERSEPROXY
+
+#include <QHostAddress>
+#include <QTcpSocket>
+#include <QObject>
+
+#include "base/http/types.h"
+
+namespace Net
+{
+    class ReverseProxy : public QObject
```

Comment:
Having separate file for so trivial function?..
Having class containing only one static method? Why? It's not C#, Java etc. It's C++!

Reply:
I know and I am neither C#,... not even C++ coder, only a plain old C coder. The function just fits nowhere else. From my point of view, proxy files were the best candidates but you don't like them.

Reply:
>only a plain old C coder. 

Then why did you decide to (mis-)use **class** if you don't understand it?

> From my point of view, proxy files were the best candidates but you don't like them.

The original place was even better for such trivial thing. But I'm still not sure it's correct for this logic...
Although, if this only works with the user's explicit agreement, it may be...